app/main.py

```diff
@@ -1,3 +1,18 @@
-def cache(func):
-    # Write your code here
-    pass
+from functools import wraps
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    results = {}
+
+    @wraps(func)
+    def inner(*args, **kwargs) -> Any:
+        key = (args, tuple(sorted(kwargs.items())))
+        if key in results:
```

Comment:
Unnecessary checking, we are receiving only args, you can then iterate over them
```suggestion
    def inner(*args) -> Any:
        if args in results:
```